File: js-stringCalculator-20131130/spec/StringCalculatorSpec.js
describe("StringCalculator", function() {

	describe("add", function() {

		it("for an empty string it will return 0", function() {
			var stringCalculator = new StringCalculator();
			var result = stringCalculator.add("");

			expect(result).toEqual(0);
		});

		it("for a string with a number it will return this number", function() {
			var stringCalculator = new StringCalculator();
			var result = stringCalculator.add("5");

			expect(result).toEqual(5);
		});

		it("for a string with multiple numbers it will return the sum of them", function() {
			var stringCalculator = new StringCalculator();
			var result = stringCalculator.add("5, 2, 3");

			expect(result).toEqual(10);
		});

		it("allows new lines as number separator", function() {
			var stringCalculator = new StringCalculator();
			var result = stringCalculator.add("5\n2");

			expect(result).toEqual(7);
		});

	});

});

File: 20190626-kotlin-string-calculator/src/test/kotlin/com/katas/StringCalculatorSpec.kt
package com.katas

import org.assertj.core.api.Assertions.assertThat
import org.jetbrains.spek.api.Spek
import org.jetbrains.spek.api.dsl.describe
import org.jetbrains.spek.api.dsl.it

object StringCalculatorSpec : Spek({

    describe("Add operation") {
        it("should return 0 when string empty") {
            assertThat(StringCalculator().add("")).isEqualTo(0)
        }

        it("should return number when string contains single number") {
            assertThat(StringCalculator().add("1")).isEqualTo(1)
        }

        it("should return sum of numbers when string contains multiple numbers") {
            assertThat(StringCalculator().add("1,2")).isEqualTo(3)
        }

        it("should return sum of numbers when string contains multiple numbers separated by new lines") {
            assertThat(StringCalculator().add("1\n2,3")).isEqualTo(6)
        }

        it("should support different delimiter") {
            assertThat(StringCalculator().add("//;\n1;2")).isEqualTo(3)
        }
    }

})